### PR TITLE
Initial draft for device binding attachments

### PR DESCRIPTION
**Pull request by paulbastian**
Signed-off-by: Paul Bastian <paul.bastian@bdr.de>

### CODE REVIEW

**Comment by paulbastian** [comment]
I updated the spec according to the feedback from Aries WG call last Wednesday

File: features/0728-device-binding-attachments/README.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    "comment": "some comment",
+    "presentations~attach": [
+        {
+            "@id": "libindy-presentation-0",
+            "mime-type": "application/json",
+            "data": {
+                "base64": "<bytes for base64>"
+            }
+        },
+        {
```
Comment (swcurran) [inline features/0728-device-binding-attachments/README.md:176]: Should be updated as with the presentation-request.

File: features/0728-device-binding-attachments/README.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+```
+
+Description of attributes:
+
+- `proofs` -- an array of proofs for different hardware keys which must match the `requests` array from the [device-binding-challenge](#device-binding-challenge)
+    - `id` -- reference to presentation of VC with an embeded hardware bound key 
+    - `path` -- JsonPath to raw value of hardware bound public key within the attached presentation of the VC represented as did:key
+    - `jws` -- Nonce from [device-binding-challenge](#device-binding-challenge) signed with the corresponding private key as a Json Web Signature object
+
+The `device-binding-response` must be attached to the `presentations~attach` array of the `presentation` message defined by [RFC-0037](https://github.com/hyperledger/aries-rfcs/blob/main/features/0037-present-proof/README.md#presentation) and [RFC-0454](https://github.com/hyperledger/aries-rfcs/tree/main/features/0454-present-proof-v2#presentation).
```
[the remark this answers is not in the document]
Comment (paulbastian) [reply features/0728-device-binding-attachments/README.md]: unfortunately auto commit did not include signoff :/

**Review by TelegramSam — APPROVED** [review]

File: features/0728-device-binding-attachments/README.md [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+Extends existing present-proof protocols to allow proofing the control of a hardware bound key embedded within a verfiable credential.
+
+## Motivation
+
+To enable use-cases which require a high level of assurance a verifier must reach a high degree of confidence that a verifiable credential (VC) can only be used by the person it was issued for. One way to enforce this requirement is that the issuer additionally binds the VC to a hardware bound public key and therefore binding the credential to the device, as discussed in the [DIF Wallet Security WG](https://github.com/decentralized-identity/wallet-security). The issaunce process, including the attestation of the wallet and the hardware bound key is off-scope for this Aries RFC. A valid presentation of the VC then requires an additional challenge which proofs that the presenter is in control of the corresponding private key.
+
+Since the proof of control must be part of legitimate presentation it makes sense to extend all current `present-proof` protocols.
+
+Note: The focus so far has been on AnonCreds, we will also look into device binding of W3C VC, however this is currently lacking in the examples.
+
```
Comment (TelegramSam) [inline features/0728-device-binding-attachments/README.md:20]: A warning here about inappropriate use of this tech?

File: features/0728-device-binding-attachments/README.md [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+        {
+            "id": "libindy-presentation-0",
+            "path": "$.requested_proof.revealed_attrs.attr1_referent.raw",
+            "jws": {
+              "header": {
+                "kid": "did:key:z6MkmjY8GnV5i9YTDtPETC2uUAW6ejw3nk5mXF5yci5ab7th"
+              },
+              "protected": "eyJhbGciOiJFZERTQSIsImlhdCI6MTU4Mzg4... (bytes omitted)",
+              "signature": "3dZWsuru7QAVFUCtTd0s7uc1peYEijx4eyt5... (bytes omitted)"
+            }
```
Comment (TimoGlastra) [inline features/0728-device-binding-attachments/README.md:142]: Would it be possible to make this a signed attachment to reuse the existing functionality?

File: features/0728-device-binding-attachments/README.md [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+    ]
+}
+```
+
+## Reference
+
+
+## Drawbacks
+
+Tbd
```
Comment (TimoGlastra) [inline features/0728-device-binding-attachments/README.md:229]: I think we could add a section here about the drawbacks in terms of privacy?

File: features/0728-device-binding-attachments/README.md (hunk 3 above)
Comment (paulbastian) [reply features/0728-device-binding-attachments/README.md:20]: done

File: features/0728-device-binding-attachments/README.md (hunk 5 above)
Comment (paulbastian) [reply features/0728-device-binding-attachments/README.md:229]: done

File: features/0728-device-binding-attachments/README.md (hunk 4 above)
Comment (paulbastian) [reply features/0728-device-binding-attachments/README.md:142]: Do you mean to move `jws` up into part of the attachment next to the `data` part of the attachment?

Comment (TimoGlastra) [reply features/0728-device-binding-attachments/README.md:142]: Yes! So it follows the spec from RFC 0017

Comment (paulbastian) [reply features/0728-device-binding-attachments/README.md]: Can you check the updated spec?

**Pull request merged**